src/morphological_operations.jl

```diff
@@ -99,3 +99,91 @@ which is defined as the arithmetic difference between the internal and the exter
 `region` allows you to control the dimensions over which this operation is performed.
 """
 morpholaplace(img::AbstractArray, region=coords_spatial(img)) = dilate(img, region) + erode(img, region) - 2img
+
+
+"""
+Applies a binary blob thinning operation, to achieve a skeletization of the input image.
+
+The function transforms a binary blob image into a skeletized form using the technique of Zhang-Suen.
+"""
+
+function thinning_iteration(img::AbstractArray, iter::Int64)
+    marker = falses(size(img))
+    h, w = size(img)
+    for i in range(2,h-1)
+        for j in range(2,w-1)
+            p2 = img[i-1][j]
+            p3 = img[i-1][j+1]
+            p4 = img[i][j+1]
+            p5 = img[i+1][j+1]
+            p6 = img[i+1][j]
+            p7 = img[i+1][j-1]
+            p8 = img[i][j-1]
+            p9 = img[i-1][j-1]
+
+            A  = (p2 == 0 && p3 == 1) + (p3 == 0 && p4 == 1) +
+                 (p4 == 0 && p5 == 1) + (p5 == 0 && p6 == 1) +
+                 (p6 == 0 && p7 == 1) + (p7 == 0 && p8 == 1) +
+                 (p8 == 0 && p9 == 1) + (p9 == 0 && p2 == 1)
+
+            B = p2 + p3 + p4 + p5 + p6 + p7 + p8 + p9
+            m1 = iter == 0 ? (p2 * p4 * p6) : (p2 * p4 * p8)
+            m2 = iter == 0 ? (p4 * p6 * p8) : (p2 * p6 * p8)
+            if (A == 1 && (B >= 2 && B <= 6) && m1 == 0 && m2 == 0)
+                marker[i,j] = 1
+            end
+        end
+    end
+    return img & ~marker            
+end    
+
+"""
+'''
+output = thinning(img)
+'''
+The function transforms a binary blob image into a skeletized form using the technique of Zhang-Suen.
+
+-  img          = Binary input image
+"""
+
+function thinning_iteration(img::AbstractArray{Bool,2}, iter::Int64)
```

Comment:
Could you please comment on this method with `Bool`? How does it differ from the general `AbstractArray` case?